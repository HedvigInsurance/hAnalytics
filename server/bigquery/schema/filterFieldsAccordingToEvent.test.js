const flattenObj = require("../flattenObj");
const filterFieldsAccordingToEvent = require("./filterFieldsAccordingToEvent");
const createBigQueryConfigMock = require("../config.mock");

test("filters according to event", async () => {
  const bigQueryConfig = createBigQueryConfigMock([
    {
      name: "mock_event",
      inputs: [
        {
          name: "input_mock",
          type: "String",
        },
      ],
    },
    {
      name: "another_mock_event",
      inputs: [
        {
          name: "store",
          type: "Dictionary<String, Any>",
        },
      ],
    },
  ]);
  const events = await bigQueryConfig.getEvents();

  await Promise.all(
    events.map(async (event) => {
      expect(
        await filterFieldsAccordingToEvent(
          event,
          flattenObj({
            context_app_name: "Hedvig",
            property: {
              store: {
                an_item: "123",
              },
            },
          }),
          bigQueryConfig
        )
      ).toMatchSnapshot(event.name);
    })
  );
});
