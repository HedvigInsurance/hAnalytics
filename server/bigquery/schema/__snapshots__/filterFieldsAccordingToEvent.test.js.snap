// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters according to event: another_mock_event 1`] = `
Object {
  "context": null,
  "event": null,
  "properties": null,
}
`;

exports[`filters according to event: mock_event 1`] = `
Object {
  "context": null,
  "event": null,
  "properties": null,
}
`;
